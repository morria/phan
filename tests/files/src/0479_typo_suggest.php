<?php

function suggest_undeclared_variable($param1, $param2) {
    $myValue = 2;
    $x = $Param2;  // should make case-insensitive suggestions

    $y = $my_value;  // should suggest changes with low levenstein distances

    $z = $notDeclaredAnywhere;

    $w = $PARAM;
}

class ExampleSuggest {
    public $_prop = 'x';
    public function example() {
        $x = $_prop;
        $prop = 2;
        $y = $_prop;
        $my_really_longname = 2;
        echo $myReallyLongname;
    }
}
