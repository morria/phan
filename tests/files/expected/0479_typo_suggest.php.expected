%s:4 PhanUnusedVariable Unused definition of variable $myValue (Did you mean $my_value)
%s:5 PhanUndeclaredVariable Variable $Param2 is undeclared (Did you mean $param2)
%s:5 PhanUnusedVariable Unused definition of variable $x
%s:7 PhanUndeclaredVariable Variable $my_value is undeclared (Did you mean $myValue)
%s:7 PhanUnusedVariable Unused definition of variable $y
%s:9 PhanUndeclaredVariable Variable $notDeclaredAnywhere is undeclared
%s:9 PhanUnusedVariable Unused definition of variable $z
%s:11 PhanUndeclaredVariable Variable $PARAM is undeclared (Did you mean $param1 or $param2)
%s:11 PhanUnusedVariable Unused definition of variable $w
%s:17 PhanUndeclaredVariable Variable $_prop is undeclared (Did you mean $this->_prop)
%s:17 PhanUnusedVariable Unused definition of variable $x
%s:18 PhanUnusedVariable Unused definition of variable $prop (Did you mean $_prop)
%s:19 PhanUndeclaredVariable Variable $_prop is undeclared (Did you mean $prop or $this->_prop)
%s:19 PhanUnusedVariable Unused definition of variable $y
%s:20 PhanUnusedVariable Unused definition of variable $my_really_longname (Did you mean $myReallyLongname)
%s:21 PhanUndeclaredVariable Variable $myReallyLongname is undeclared (Did you mean $my_really_longname)
